### DISCUSSION TITLE
Joint's child frame visualization

### DISCUSSION THREAD

**Discussion by mrrezaie** [ComputationalBiomechanicsLab/opensim-creator#972] [Q&A] [answered]
Hi, 

Happy new year,

I was trying to understand how the orientation of parent and child frames should be, and found that the visualizations of the axes in OSC and OpenSim are not identical. 

This is the `talus_r_offset` frame of the `ankle_r` joint in Rajagopal2016 model in OpenSim:
![image](https://github.com/user-attachments/assets/b4865fb8-7b5d-464b-96c1-d7b9ba5d13d9)

And the same frame in OSC (notice that it is perfectly orthogonal):
![image](https://github.com/user-attachments/assets/6f86b8a4-181f-4109-a977-dde4deff42de)

I was wondering if you could help me understand which one is correct.

Thanks

**Comment by adamkewley** [comment, marked as answer]
That visualization is not the frame, it's the editor gizmo. Because the location of the frame is fixed by the joint contrtaint, the editor gizmo ends up showing you where the *parent* of the child is (because that's effectively what is edited by editing that offset frame). Issue #955 explains this in more detail.

To toggle the visualization of frames (again, not the 3D user gizmo manipulators, but I agree they have similar visual representations), there's a button in the top toolbar that looks like a frame. This toggles showing all frames (clarifying it's existence is issue #954, making it possible to only show the current selection is issue #956). 

Apologies for the confusion. Welcome to user interface design: VFX designers understand that the arrow manipulators are for manipulating objects in 3D space but mechanical engineers and OpenSim GUI users are accustomed to seeing RGB lines as being a representation of a coordinate system origin. Subtle changes to UI design, button placement, color choices, etc. can help alleviate issues like that (but it takes time, iteration, etc.)

The button you're looking for is the leftmost of these:

![image](https://github.com/user-attachments/assets/d50e8784-c63c-40b6-b04d-e84b843c19c4)

**Comment by adamkewley** [comment]
a2edc7ed8 adds the buttons to the `Display` menu

**Comment by mrrezaie** [comment]
@adamkewley, thanks a lot for the clarification, now I have a good understanding about it. (I should have posted this in the Discussions, sorry)
